File: AOJ/0418/laver.py
MAX_X = 2000
MAX_Y = 2000
s = [ [0 for i in range(MAX_X)] for j in range(MAX_Y) ]
for i in range (2) :
    x,y,w,h = map(int,input().split())
    for i in range(x, x + w) :
        for j in range(y, y + h) :
            s[i][j] += 1
c = 0
for i in range(0, MAX_X) :
    for j in range(0, MAX_Y) :
        if s[i][j] == 1 : c += 1
print(c)

File: AOJ/0418/laver.cpp
#include<iostream>
using namespace std;
const int MAX_X = 2000;
const int MAX_Y = 2000;

int grid[MAX_X][MAX_Y]; // 海苔が置かれる区画。 グローバル変数のため0に初期化される。

int main() {
  int x, y, w, h;
  for ( int k = 0; k<2; ++k ) {
    cin >> x >> y >> w >> h;
    for ( int i = x; i < x + w; ++i ) 
      for ( int j = y; j < y + h; ++j ) ++grid[i][j];
  }
    
  int c = 0;
  for ( int i = 0; i < MAX_X; ++i ) 
    for ( int j = 0; j < MAX_Y; ++j ) 
      if ( grid[i][j] == 1 ) ++c;
  cout << c << endl;
  return 0;
}

